### PR TITLE
Add testing GitHub Actions workflow

### PR DESCRIPTION
**Pull request by schlessera**
Adds a GitHub Actions workflow for running both unit and functional tests.

### CODE REVIEW

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+          MYSQL_ALLOW_EMPTY_PASSWORD: yes
+        ports:
+          - 3306:3306
+        options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3
+
+    steps:
+      - name: Check out source code
+        uses: actions/checkout@v2
+
+      - name: Check existence of composer.json & phpcs.xml.dist files
```
Comment (swissspidy) [inline .github/workflows/testing.yml:65]: Description doesn't match `files` below

**Comment by swissspidy** [comment]
Might wanna remove those HTML files

File: .github/workflows/testing.yml [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-composer-
+
+      - name: Install dependencies
+        if: steps.check_files.outputs.files_exists == 'true'
+        run: COMPOSER_ROOT_VERSION=dev-master composer install --prefer-dist --no-progress --no-suggest
+
+      - name: Prepare test database
+        run: composer prepare-tests
```
Comment (swissspidy) [inline .github/workflows/testing.yml:100]:
```
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)
```

Might need to expose the right env vars / ports? In my project I use `${{ job.services.mysql.ports['3306'] }}` as the port, if that helps.

File: .github/workflows/testing.yml [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+      - name: Install dependencies
+        if: steps.check_files.outputs.files_exists == 'true'
+        run: COMPOSER_ROOT_VERSION=dev-master composer install --prefer-dist --no-progress --no-suggest
+
+      - name: Show journalctl status
+        run: sudo journalctl -xe
+
+      - name: Prepare test database
+        run: MYQSL_HOST=127.0.0.1 MYSQL_TCP_PORT=${{ job.services.mysql.ports['3306'] }} composer prepare-tests
```
Comment (wojsmol) [inline .github/workflows/testing.yml:102]: Start MySQL before this steep  - see [this Github blog post](https://github.blog/changelog/2020-02-21-github-actions-breaking-change-ubuntu-virtual-environments-will-no-longer-start-the-mysql-service-automatically/)

**Pull request merged**